chore(savedaddresses)!: favourite property removed and primary key updated

saledjenic opened the pull request:
- favourite column removed from the saved_addresses table
- favourite property removed from the SavedAddress struct
- ens name removed from the primary key, the primary key now is composed of address and is_test columns
- ens parameter removed from wakuext_deleteSavedAddress
- wallet_getSavedAddresses moved to wakuext_getSavedAddresses (to keep them all in a single place)
- saved addresses related endpoints removed from the wallet service, even they logically belong there, a reason for that
is avoiding emitting sync message if one uses calls from the wallet service, while that's not the case in ext service. Once
we refactor this and introduce devices syncing mechanism in the wallet service, we should not only these but other wallet
related endpoints move there (removed: wallet_getSavedAddresses, wallet_addSavedAddress and wallet_deleteSavedAddress).

Affected area:
Saved addresses

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (16)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 56fd4a22 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/1/) | 2024-01-08 10:28:14 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-102441-56fd4a-pr4537.zip) |
| :heavy_multiplication_x: | 56fd4a22 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/1/) | 2024-01-08 10:28:56 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/1//consoleText) |
| :heavy_check_mark: | 56fd4a22 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4537/1/) | 2024-01-08 10:30:23 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-102441-56fd4a-pr4537.aar) |
| :heavy_check_mark: | 56fd4a22 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/1/) | 2024-01-08 10:30:44 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-102441-56fd4a-pr4537.zip) |
| | | | | | | |
| :heavy_check_mark: | 0321102c | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4537/2/) | 2024-01-08 11:16:23 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-111341-032110-pr4537.aar) |
| :heavy_check_mark: | 0321102c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/2/) | 2024-01-08 11:16:54 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-111341-032110-pr4537.zip) |
| :heavy_check_mark: | 0321102c | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/2/) | 2024-01-08 11:17:06 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-111341-032110-pr4537.zip) |
| :heavy_multiplication_x: | 0321102c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/2/) | 2024-01-08 11:17:51 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/2//consoleText) |
| :heavy_multiplication_x: | 0321102c | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/3/) | 2024-01-08 11:35:35 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 25976f3a | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4537/3/) | 2024-01-08 11:49:22 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-114546-25976f-pr4537.aar) |
| :heavy_check_mark: | 25976f3a | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/3/) | 2024-01-08 11:50:53 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-114552-25976f-pr4537.zip) |
| :heavy_check_mark: | 25976f3a | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/3/) | 2024-01-08 11:55:52 | ~10 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-114552-25976f-pr4537.zip) |
| :heavy_check_mark: | 25976f3a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/4/) | 2024-01-08 12:26:18 | ~40 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 64b6d6e0 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/4/) | 2024-01-08 15:00:54 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-145714-64b6d6-pr4537.zip) |
| :heavy_check_mark: | 64b6d6e0 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/4/) | 2024-01-08 15:00:57 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-145717-64b6d6-pr4537.zip) |
| :heavy_multiplication_x: | 64b6d6e0 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/5/) | 2024-01-08 15:29:34 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/5//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a998ef0f | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4537/5/) | 2024-01-09 15:03:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240109-150110-a998ef-pr4537.aar) |
| :heavy_check_mark: | a998ef0f | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/5/) | 2024-01-09 15:04:21 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240109-150110-a998ef-pr4537.zip) |
| :heavy_check_mark: | a998ef0f | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/5/) | 2024-01-09 15:04:39 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240109-150110-a998ef-pr4537.zip) |
| :heavy_multiplication_x: | a998ef0f | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/6/) | 2024-01-09 15:16:28 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/6//consoleText) |
| :heavy_multiplication_x: | a998ef0f | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/7/) | 2024-01-09 16:43:46 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 939eb5de | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4537/6/) | 2024-01-10 07:26:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240110-072514-939eb5-pr4537.zip) |
| :heavy_check_mark: | 939eb5de | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4537/6/) | 2024-01-10 07:27:23 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240110-072510-939eb5-pr4537.aar) |
| :heavy_check_mark: | 939eb5de | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4537/6/) | 2024-01-10 07:28:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240110-072514-939eb5-pr4537.zip) |
| :heavy_multiplication_x: | 939eb5de | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/8/) | 2024-01-10 07:45:31 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/8//consoleText) |
| :heavy_check_mark: | 939eb5de | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/9/) | 2024-01-10 09:22:05 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4537/9//consoleText) |

saledjenic committed:
chore(savedaddresses)!: favourite property removed and primary key updated

- favourite column removed from the saved_addresses table
- favourite property removed from the SavedAddress struct
- ens name removed from the primary key, the primary key now is composed of address and is_test columns
- ens parameter removed from wakuext_deleteSavedAddress
- wallet_getSavedAddresses moved to wakuext_getSavedAddresses (to keep them all in a single place)
- saved addresses related endpoints removed from the wallet service, even they logically belong there, a reason for that
is avoiding emitting sync message if one uses calls from the wallet service, while that's not the case in ext service. Once
we refactor this and introduce devices syncing mechanism in the wallet service, we should not only these but other wallet
related endpoints move there (removed: wallet_getSavedAddresses, wallet_addSavedAddress and wallet_deleteSavedAddress).

Affected area:
Saved addresses

ilmotta commented:
cc @J-Son89 I'm not sure about the new wallet in mobile, but these upcoming breaking changes seem to at least affect our legacy wallet.

stefandunca commented on services/ext/api.go:
```diff
@@ -1066,8 +1066,12 @@ func (api *PublicAPI) UpsertSavedAddress(ctx context.Context, sa wallet.SavedAdd
 
-func (api *PublicAPI) DeleteSavedAddress(ctx context.Context, address ethcommon.Address, ens string, isTest bool) error {
-	return api.service.messenger.DeleteSavedAddress(ctx, address, ens, isTest)
+func (api *PublicAPI) DeleteSavedAddress(ctx context.Context, address ethcommon.Address, isTest bool) error {
```
It's strange to see the saved address APIs here. However, I assume this is due to our current architecture.

alaibe replied:
This is to have access to waku :(

saledjenic replied:
> It's strange to see the saved address APIs here. However, I assume this is due to our current architecture.

Yes, it is, I agree, but having Add/Delete saved address endpoints directly exposed as a part of wallet services avoids sending sync messages when the client uses them and that's why I removed them from there and kept them only here, to avoid misleading usage.

Logically those calls belong to the wallet service, which means once we establish an architecture we should move them there, but till then we should have them in a single place.

stefandunca commented on walletdatabase/migrations/sql/1704701942_remove_favourite_and_change_primary_key_for_saved_addresses.up.sql:
```diff
@@ -0,0 +1,42 @@
+  created_at INT NOT NULL DEFAULT 0,
+  color VARCHAR NOT NULL DEFAULT "primary",
+  PRIMARY KEY (address, is_test),
+  UNIQUE (name, is_test)
```
I see test mode polluting more of the functionality by adding extra responsibilities, I'm thinking that we should consider abstracting it at a higher level until it is not too entangled. My first thought is that having a separate wallet DB for testing might make it easier.

saledjenic replied:
Not sure if you're pointing to status-go unit/integration tests or test mode, in this example `is_test` field is used to mark the addresses that are saved when the app is in the test mode. 

Not entirely sure up to which level is the test mode entangled in the app at this moment, like do we differentiate the data for all wallet-related stuff based on the test mode, but if we do, that's a valid suggestion for sure.

stefandunca replied:
> Not sure if you're pointing to status-go unit/integration tests or test mode, in this example is_test field is used to mark the addresses that are saved when the app is in the test mode.

I was referring to the blockchain test mode (where we use the chains instead of .

> Not entirely sure up to which level is the test mode entangled in the app at this moment, like do we differentiate the data for all wallet-related stuff based on the test mode, but if we do, that's a valid suggestion.

We have to use it for all user state-related data that should not be mixed between main/test modes (usually chain-related like in this case). We should keep an eye on it though if it grows too much.

stefandunca reviewed (approved):
Good job. I left some brainstorming ideas in the comments.

J-Son89 commented:
> cc @J-Son89 I'm not sure about the new wallet in mobile, but these upcoming breaking changes seem to at least affect our legacy wallet.

the legacy wallet is removed at this point so it should be okay. I'm sure we will communicate if any changes should affect the current mobile wallet 👌
